omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -0,0 +1,225 @@
+	}
+	
+	private void restartModules(Collection<Module> modules, ServletContext servletContext) {
+		stopModules(modules, servletContext, false);
+		startModules(modules, servletContext);
+	}
+	
+	private void unloadModules(Collection<Module> modules, ServletContext servletContext) {
+		boolean needsRefresh = false;
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+				needsRefresh = true;
+			}
+			moduleFactoryWrapper.unloadModule(module);
+		}
+		
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	/**
+	 * @param modules modules to stop
+	 * @param servletContext ServletContext is required by WebModuleUtil to perform operation
+	 */
+	private void stopModules(Collection<Module> modules, ServletContext servletContext, boolean refreshContext) {
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+			}
+		}
+		if (refreshContext) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	/**
+	 * @param modules modules to start
+	 * @param servletContext ServletContext is required by WebModuleUtil to perform operation
+	 */
+	private void startModules(Collection<Module> modules, ServletContext servletContext) {
+		boolean needsRefresh = false;
+		if (modules.size() > 1) {
+			modules = moduleFactoryWrapper.getModulesInStartupOrder(modules);
+		}
+		
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStopped(module)) {
+				needsRefresh = moduleFactoryWrapper.startModuleSkipRefresh(module, servletContext) || needsRefresh;
+			}
+		}
+		//check if any module has been started, doesn't refresh WAC if all modules failed to start
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+		
+		findAndThrowStartupErrors(modules);
+	}
+	
+	private void findAndThrowStartupErrors(Collection<Module> modules) {
+		List<Exception> errors = new ArrayList<Exception>();
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStopped(module)) {
+				//module actions are executed in other thread, so we need to explicitly check and throw them
+				if (module.getStartupErrorMessage() != null) {
+					errors.add(new ModuleException(module.getStartupErrorMessage()));
+				}
+			}
+		}
+		
+		if (!errors.isEmpty()) {
+			StringBuilder stringBuilder = new StringBuilder();
+			for (Exception error : errors) {
+				stringBuilder.append(error.getMessage()).append("; ");
+			}
+			throw new RuntimeException(stringBuilder.toString());
+		}
+	}
+	
+	@Override
+	public ModuleAction newDelegate() {
+		return new ModuleAction();
+	}
+	
+	@Override
+	public ModuleAction save(ModuleAction delegate) {
+		throw new UnsupportedOperationException("ModuleAction cannot be saved");
```

Comment:
I'm overwriting `create` method because to execute any operation on module, ServletContext has to be passed, it is commented in code: https://github.com/PawelGutkowski/openmrs-module-webservices.rest/blob/460018808b7bbff61d3405ea916346b1370b84c8/omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java#L49